src/alfasim_sdk/_internal/alfacase/migration.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any
+
+from strictyaml import YAML
+from strictyaml.parser import generic_load
+
+from alfasim_sdk._internal.alfacase.case_description_attributes import DescriptionError
+
+
+def migrate_alfacase_yaml_to_latest(yaml_contents: str) -> str:
+    """
+    Migrates the given YAML contents of an alfacase file to the latest schema version.
+
+    We follow an approach similar to database migration, where we change the underlying data
+    directly before attempting to load it using the schema.
+
+    This function loads the yaml data as a `dict` and passes it on to various functions that are
+    responsible for migrating one aspect that has changed in the `alfacase` file since its first
+    version, which can then be loaded by the Schema directly.
+
+    Returns the updated YAML which can be handed over to the schema functions.
+    """
+    # generic_load does not raise error when try to read an invalid yaml text based.
+    yaml_data = generic_load(yaml_contents, allow_flow_style=True)
+    if isinstance(yaml_data.data, str):
+        raise DescriptionError(
+            f"When expecting a mapping found arbitrary text: {yaml_data.data}"
+        )
```

Comment:
If you expect a dict, it makes more sense to test for that, not test for some other possibility:


```suggestion
    if not isinstance(yaml_data.data, dict):
        raise DescriptionError(
            f"Expected a mapping for the alfacase file, but got: {yaml_data.data!r}"
        )
```